### ISSUE TITLE
Export Process: Delimiter change and new export

### ISSUE THREAD

**Issue by ajparsons** [mysociety/alaveteli#4527]
I need a new copy of the WDTK export that covers last year - but I'd also like the delimiter to change (at least for the incomingmessage file, but also changing all is fine) - message content is spilling over at the moment. Tab-delimited is probably fine but better suggestions also good.

**Comment by garethrees** [comment]
Hmm, the exporter _should_ handle quotes in the content. Will take a look.

**Comment by garethrees** [comment]
Had some errors when this was just run. Lost the scrollback now, but here's the snippet I got:

```
/opt/rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:22:in `dispatch'
/opt/rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/opt/rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:13:in `start'
/opt/rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle:30:in `block in <top (required)>'
/opt/rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
/opt/rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle:22:in `<top (required)>'
/opt/rbenv/versions/2.3.5/bin/bundle:22:in `load'
/opt/rbenv/versions/2.3.5/bin/bundle:22:in `<main>'
[326937, 170748, Thu, 16 Jan 2014 16:00:26 GMT +00:00, Thu, 16 Jan 2014 16:00:26 GMT +00:00]
""
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '??'
lib/data_export.rb:74: warning: regular expression has redundant nested repeat operator '*': /Mr Br***er/
```

The gist of it was that in names like e.g. `Bob ( smith` the bracket was interpreted as part of the regexp rather than escaped.

We were ending up doing `handle_error`: https://github.com/mysociety/alaveteli/blob/develop/lib/data_export.rb#L173-L180

**Comment by lizconlan** [comment]
Name formats that contain characters which are meaningful to regular expression parsers are causing most of these problems e.g. (fake data based on real export failures)

`Liz (Test account)`

`*name removed* (Account suspended)` (both the `*` and the `()` potentially problematic there, it's failing the 'repeat operator' before it finds the parentheses)

Ironically in the second case as the failure comes from a routine that's using a regex while attempting to anonymise the data - https://github.com/mysociety/alaveteli/blob/develop/lib/data_export.rb#L80-L89 which ends up https://github.com/mysociety/alaveteli/blob/develop/lib/data_export.rb#L74 `text.gsub(/#{user.name}/i, "<REQUESTER>")`

**Comment by garethrees** [comment]
```diff
- text.gsub(/#{user.name}/i, "<REQUESTER>")
+text.gsub(/#{ Regexp.escape(user.name) }/i, "<REQUESTER>")
```

^ should do it

**Comment by lizconlan** [comment]
Regex bug fixed by #4981

**Comment by garethrees** [comment]
> There is also a (transient?) bug when exporting the User data and anonymising it with the gender_detector gem. This might be fixed by upgrading it

We updated `gender_detector` in https://github.com/mysociety/alaveteli/commit/2158b693c9f34e3a9c73714dddf359444f39dc46 so closing assuming this is fixed. Can reopen or create new issue for any future problems.
